compiler/rustc_expand/src/proc_macro.rs

```diff
@@ -74,6 +79,7 @@ impl MultiItemModifier for ProcMacroDerive {
         _meta_item: &ast::MetaItem,
         item: Annotatable,
     ) -> ExpandResult<Vec<Annotatable>, Annotatable> {
+        let _timer = ecx.sess.prof.generic_activity_with_arg("expand_proc_macro", self.name);
```

Comment:
Do you primarily want to profile user code?
Then I'd suggest to wrap only `client.run(...)` into the profiling function.
Otherwise you get times from some pretty arbitrary parts of the expansion infra as well (these parts are also different for e.g. fn-like macros and derive macros).

Reply:
I do want to indeed profile user-code to have more visibility into sources of slowness here, and e.g. a way to pinpoint some slow proc-macros (so that users can try to change their code or open issues, or have cargo build these slow proc-macros in release mode at the expense of slower compile times, etc). 

I'll switch to `client.run` then, thank you. I believe that the current way this PR profiles expansions does reach all expansion cases/all situations (for example, I think I didn't see this logging happening in dependencies of the current crate ?)

Reply:
I've limited the profiling's scope to the `client.run()` call so it'll be more accurate. 

Ensuring more paths are covered by the self-profiler is definitely interesting, but I can do that in followups: just seeing some expansions like here is already good information.